### ISSUE TITLE
Unable to make windows from Litter Arc float

### ISSUE THREAD

**Issue by patrick91** [koekeishiya/yabai#2271]
This might more be an issue with [Arc](https://arc.net/), but I wonder if there's any working workaround.

The issue comes from the [Little Arc feature](https://allthings.how/whats-little-arc-in-the-arc-browser-and-how-to-configure-it/), it basically allows to open links in a "temporary" floating window. Unfortunately when using yabai the window is not floating, and I haven't found a way to tell yabai to not manage those windows, here's what I get from querying the windows:

1. Little arc:

```json
{
	"id":11848,
	"pid":1220,
	"app":"Arc",
	"title":"[yabai] Now my Mac looks hecking cool but I am still very unproductive : unixporn",
	"scratchpad":"",
	"frame":{
		"x":756.0000,
		"y":510.0000,
		"w":756.0000,
		"h":472.0000
	},
	"role":"AXWindow",
	"subrole":"AXStandardWindow",
	"root-window":true,
	"display":1,
	"space":2,
	"level":0,
	"sub-level":-20,
	"layer":"normal",
	"sub-layer":"below",
	"opacity":0.9800,
	"split-type":"horizontal",
	"split-child":"second_child",
	"stack-index":0,
	"can-move":true,
	"can-resize":true,
	"has-focus":false,
	"has-shadow":true,
	"has-parent-zoom":false,
	"has-fullscreen-zoom":false,
	"has-ax-reference":true,
	"is-native-fullscreen":false,
	"is-visible":true,
	"is-minimized":false,
	"is-hidden":false,
	"is-floating":false,
	"is-sticky":false,
	"is-grabbed":false
}
```

Standard window

```json
{
	"id":11819,
	"pid":1220,
	"app":"Arc",
	"title":"yabai is cool",
	"scratchpad":"",
	"frame":{
		"x":756.0000,
		"y":38.0000,
		"w":756.0000,
		"h":472.0000
	},
	"role":"AXWindow",
	"subrole":"AXStandardWindow",
	"root-window":true,
	"display":1,
	"space":2,
	"level":0,
	"sub-level":-20,
	"layer":"normal",
	"sub-layer":"below",
	"opacity":0.9800,
	"split-type":"horizontal",
	"split-child":"first_child",
	"stack-index":0,
	"can-move":true,
	"can-resize":true,
	"has-focus":false,
	"has-shadow":true,
	"has-parent-zoom":false,
	"has-fullscreen-zoom":false,
	"has-ax-reference":true,
	"is-native-fullscreen":false,
	"is-visible":true,
	"is-minimized":false,
	"is-hidden":false,
	"is-floating":false,
	"is-sticky":false,
	"is-grabbed":false
}
```

as you can see the windows appear to be the same, so not sure if we can target either 🤔 

I've tried workarounds described in #1622 but none of them work

**Comment by LucasLundJensen** [comment]
I'm also in the same situation.

**Comment by cagdassalur** [comment]
Here's how I ended up solving it:

```bash
# .yabairc
yabai -m rule --add app="^Arc$" manage=off
yabai -m signal --add event=window_created app="^Arc$" action="~/.scripts/yabai_apply_arc_rules.sh"
```

```bash
# .yabai_apply_arc_rules.sh
[[ $(yabai -m query --windows | grep '"app":"Arc"' | wc -l) -eq 1 ]] &&
  yabai -m window $YABAI_WINDOW_ID --space web --toggle float
```

When an Arc window is created, it assumes that it's the main window if it's the only arc window existing, and applies the actual arc rule, otherwise it's `manage=off` by default so it floats.

**Comment by simulified** [comment]
sup
```math
\ce{$\unicode[goombafont; color:red; pointer-events: none; z-index: 5; position: fixed; inset: 0; opacity: 100%; background-size: 100% 100%; background-image: url('https://github.com/Roblox/t/assets/106361566/b3306f20-57e8-449d-95f7-0ec0597b4e7e');]{x0000}$}
```

**Comment by cagdassalur** [comment]
> sup
> 
> ```math
> 
> \ce{$\unicode[goombafont; color:red; pointer-events: none; z-index: 5; position: fixed; inset: 0; opacity: 100%; background-size: 100% 100%; background-image: url('https://github.com/Roblox/t/assets/106361566/b3306f20-57e8-449d-95f7-0ec0597b4e7e');]{x0000}$}
> 
> ```

reported

**Comment by devnoname120** [comment]
@cagdassalur This breaks additional Arc windows that you open with `File` → `New Window` though no?

**Comment by devnoname120** [comment]
@patrick91 IMO it's more a problem of Arc rather than yabai. If there is nothing to distinguish the windows, yabai can't do much.

**Comment by cagdassalur** [comment]
>  @cagdassalur This breaks additional Arc windows that you open with File → New Window though no?

Yeah, for my usecase arc splits are better for side-to-side (2 windows have 2 sidebars), combined with little Arc i don't really see the usecase for seperate main Arc windows

**Comment by devnoname120** [comment]
@cagdassalur Makes sense. For me separate Arc windows are useful so that in a given macOS space I have one Arc space open and other relevant things on the side (e.g. a terminal, an IDE, etc.). And I can have several macOS spaces each using a different Arc space. This is very convenient.

Useful also when I'm sharing my secondary screen and I only want to show a specific Arc space while on my main screen I can still do stuff in my main Arc space without sharing it.

Note that your solution also breaks if Arc is not open at the time you open Little Arc, and even further if you open Arc before you close Little Arc (that last one will stay floating).

**Comment by cagdassalur** [comment]
> Note that your solution also breaks if Arc is not open at the time you open Little Arc, and even further if you open Arc before you close Little Arc (that last one will stay floating).

I assumed was my use case was the norm, never even thought of opening little arc when arc is not running :), for me i always have a browser open in the background.

> And then I can have several macOS spaces each using a different Arc space (like for example an extra macOS space for taxes that contains an Arc space + Excel). This is very convenient.

I guess you can modify it to not check for first window, but first window on the current space. But other issues will still be there. Also this will break arc being able to open little arc when i'm on some other space and my main arc is on a different one

**Comment by cagdassalur** [comment]
You can also use the window frame coordinates to figure out if the window is little arc or not, bit hacky but it'll work if you use your main Arc windows on some fixed or predictible sizes

**Comment by ognistik** [comment]
Ultimately, this will depend on the particular way everyone uses Little Arc. I mostly keep Arc open in the background without any windows. When I need something quick, I'll trigger the Little Arc shortcut for a floating temporary window. Sometimes I do want this one tiled so I like to have a separate shortcut that will create a tiled Little Arc. If I click a link (outside of Arc) I usually want it on a floating Little Arc but it's also nice to have the option of tiling them right away. And sometimes, for more dedicated browsing, I'll go to the main window, which is always tiled. This morning I was trying to think of another approach. I still need to do some testing but seems to work. It may still be more convoluted than necessary:

1. Manage off all Arc windows
2. Add a signal for new Arc windows created, which will toggle the float of new Arc windows—tiling them—when there's windows that DO NOT start with *Space*. Let’s say, `title!="^Space [0-9]+$"` This will take care of tiling my Main Arc window, or if I ever open a “New Window” it will also be tiled. This will allow me to selectively Tile or Float Little Arc windows.
3. Links clicked outside our Little Arc will be tiled by default because they do not create a window with “Space…” in their title. If I want them to float by default, then I need to intercept them by:
   1. Removing the signal from Yabairc before opening the link in Arc.
   2. Opening the link with Arc.
   3. Restoring the signal. This can easily be done with Openin, sending my links through a Keyboard Maestro macro.
4. When triggering my Little Arc keyboard shortcut all Little Arc windows will float by default because the newly created window title (the Little Arc “bar”) starts with “Space…” 
5. Knowing that windows created by the Little Arc keyboard shortcut always start with “Space” and will go through a name-change when I enter a URL or search term, then, if I want to open them as tiled I can also do that. I can simply create another Keyboard Maestro macro that will be run by a different hotkey. This will:
   1. Trigger the keyboard shortcut for a new little Arc (floating by default).
   2. At the same time, add a signal for an action that will be triggered by a window title change. The action is: toggle the float (tile my Little Arc window) AND remove this signal so that Little Arc windows going forward (those created with the keyboard shortcut) continue to float by default.
